### PR TITLE
add module gitlab_branch

### PR DESCRIPTION
**Pull request by paytroff**
##### SUMMARY
- add module gitlab-branch to create/delete branch in project gitlab
- 
##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME
gitlab_branch

### CODE REVIEW

File: plugins/modules/source_control/gitlab/gitlab_branch.py
Comment (felixfontein) [inline plugins/modules/source_control/gitlab/gitlab_branch.py:15]:
```suggestion
  - Werner Dijkerman (@dj-wasabi)
  - You have to add yourself here
```

File: plugins/modules/source_control/gitlab/gitlab_branch.py
Comment (felixfontein) [inline plugins/modules/source_control/gitlab/gitlab_branch.py:11]:
```suggestion
short_description: Create or delete a branch
version_added: 4.2.0
```

File: plugins/modules/source_control/gitlab/gitlab_branch.py
Comment (felixfontein) [inline plugins/modules/source_control/gitlab/gitlab_branch.py:13]: Descriptions (except short_descriptions) should always contain full sentences (or multiple full sentences).
```suggestion
  - This module allows to create or delete branches.
```

File: plugins/modules/source_control/gitlab/gitlab_branch.py
Comment (felixfontein) [inline plugins/modules/source_control/gitlab/gitlab_branch.py:46]:
```suggestion
      - Reference branch to create from.
      - This must be specified if I(state=present).
```

File: plugins/modules/source_control/gitlab/gitlab_branch.py
Comment (felixfontein) [inline plugins/modules/source_control/gitlab/gitlab_branch.py:144]: Adding support for check mode is rather simple, this is something you should really do. Users often expect that check mode works.

**Review by felixfontein — COMMENTED** [review]
As mentioned before, this needs tests.

File: plugins/modules/source_control/gitlab/gitlab_branch.py
Comment (paytroff) [reply plugins/modules/source_control/gitlab/gitlab_branch.py:146]: You are right, next time. I'll do it.

File: plugins/modules/source_control/gitlab/gitlab_branch.py
Comment (paytroff) [reply plugins/modules/source_control/gitlab/gitlab_branch.py:16]: tests added.

Comment (felixfontein) [reply plugins/modules/source_control/gitlab/gitlab_branch.py:16]: You still need to add yourself here.

File: plugins/modules/source_control/gitlab/gitlab_branch.py
Comment (felixfontein) [reply plugins/modules/source_control/gitlab/gitlab_branch.py:145]: Why not for this PR? :)

File: tests/integration/targets/gitlab_branch/aliases
Comment (felixfontein) [inline tests/integration/targets/gitlab_branch/aliases:2]:
```suggestion
```
This is not a valid directive.

File: plugins/modules/source_control/gitlab/gitlab_branch.py
Comment (paytroff) [reply plugins/modules/source_control/gitlab/gitlab_branch.py:145]: I can't.

File: plugins/modules/source_control/gitlab/gitlab_branch.py
Comment (paytroff) [reply plugins/modules/source_control/gitlab/gitlab_branch.py:16]: I can't, i'm not in maintainers list. I have an error with checks when I do it.

Comment (felixfontein) [reply plugins/modules/source_control/gitlab/gitlab_branch.py:16]: You need to add yourself to the maintainers list for this module as well.

File: plugins/modules/source_control/gitlab/gitlab_branch.py
Comment (felixfontein) [reply plugins/modules/source_control/gitlab/gitlab_branch.py:145]: Why not?

Comment (paytroff) [reply plugins/modules/source_control/gitlab/gitlab_branch.py:145]: Not enough time.

File: plugins/modules/source_control/gitlab/gitlab_branch.py
Comment (paytroff) [reply plugins/modules/source_control/gitlab/gitlab_branch.py:16]: How do that?

Comment (felixfontein) [reply plugins/modules/source_control/gitlab/gitlab_branch.py:16]: Simply add a new entry like https://github.com/ansible-collections/community.general/blob/main/.github/BOTMETA.yml#L950-L951 for the new module.

**Review by felixfontein — APPROVED** [review]
Looks good to me!

If nobody reviews until beginning of next week, I'll merge.

**Comment by felixfontein** [comment]
@paytroff thanks for your contribution!
@sh0shin thanks for reviewing!

**Pull request merged**
